index.html

```diff
@@ -487,6 +495,14 @@ <h3>Proofs</h3>
 <em>proof value</em> generated by the
 <a href="#generate-proof">Generate Proof Algorithm</a>.
           </dd>
+
+          <dt>previousProof</dt>
+          <dd>
+An optional string value that identifies another <a>data integrity proof</a>
+that MUST verify before the current proof is processed. This property is used
```

Comment:
The order of verification seems backward. I would think an enclosing proof should be verified before an enclosed (previous) proof. Likewise for enclosing encryption before enclosed (previous) encryption... I may be conflating encryption and signing/proving, which I think means we need to clarify how these will/do interoperate.
```suggestion
that MUST verify before the current proof is processed. This property is used
```

Reply:
> I may be conflating encryption and signing/proving

Yes, this purely has to do with signing and not encryption.

Fundamentally, with chained signatures, you need to verify the entire chain so from that perspective, it doesn't really matter where you start.

One argument for where one should prefer to start is to try and determine where an attacker would likely strike if they were trying to DoS you. Presuming you need to fetch some of these keys from the network (like if people are using `did:web` or any ledger-based DID Method), an attacker could attack you by providing a bunch of bogus keys in a chain of proofs. So, of that chain, where are they most likely to attack? 

Would an attacker choose to provide a chain where the attack happens at the beginning or the end of a chain. That might suggest where we should start verification.

Fundamentally, if the first proof in a chain is invalid, the rest of the chain is invalid. If the last proof in a chain is invalid, that might be ok from a verifier perspective... remember, with the `previousProof` property... you can have directed acyclic graphs of signatures, where only a subset might need to verify for the Verifier to have confidence in the signed information. Now, I know that this PR does not talk about any of that... and those use cases are experimental (and we probably won't try to standardize them for years)... but when we get there, it hints at first-to-last verification order having a slight (theoretical) advantage over last-to-first. Maybe what we want is good caching behaviour, where we cache as much as we can early on, and bail as early on in the chain as we can?

So, I think the current argument is: If we do first to last, or last to first verification doesn't matter... and we'll put language in there that says that as long as the result of an alternate algorithm is the same as the one documented in the spec, it's conformant (which is typical language to put in an algorithms section).

The future argument might be: We optimize for caching the signature check of the head of the chain, because we might have multi-branching in these proof chains (effectively, proof trees) and so we should ensure that the root of the proof tree is valid before continuing down the branches. 

In any case, even if we were to pick the wrong thing here, I don't think it prevents us from changing it in the future in a way that doesn't cause backwards-incompatible changes.

... and to assuage people's concerns, we might want to start out with very strong "every proof in the chain MUST verify" language and then weaken it later once people provide good examples for why it's ok for some proofs in the chain to not verify.